DuckDuckGo/Configuration/ConfigurationStore.swift

```diff
@@ -93,24 +158,46 @@ final class ConfigurationStore: ConfigurationStoring {
 
     func loadData(for config: Configuration) -> Data? {
         let file = fileUrl(for: config)
-        do {
-            return try Data(contentsOf: file)
-        } catch {
-            guard NSApp.runType.requiresEnvironment else { return nil }
+        var data: Data?
+        var coordinatorError: NSError?
+
+        NSFileCoordinator().coordinate(readingItemAt: file, error: &coordinatorError) { fileUrl in
+            do {
+                data = try Data(contentsOf: fileUrl)
+            } catch {
+                guard NSApp.runType.requiresEnvironment else { return }
 
-            let nserror = error as NSError
+                let nserror = error as NSError
 
-            if nserror.domain != NSCocoaErrorDomain || nserror.code != NSFileReadNoSuchFileError {
-                PixelKit.fire(DebugEvent(GeneralPixel.trackerDataCouldNotBeLoaded, error: error))
+                if nserror.domain != NSCocoaErrorDomain || nserror.code != NSFileReadNoSuchFileError {
+                    PixelKit.fire(DebugEvent(GeneralPixel.trackerDataCouldNotBeLoaded, error: error))
+                }
             }
+        }
 
-            return nil
+        if let coordinatorError {
+            PixelKit.fire(DebugEvent(GeneralPixel.configurationFileCoordinatorError, error: coordinatorError))
+            Logger.config.error("Unable to read \(config.rawValue, privacy: .public): \(coordinatorError.localizedDescription, privacy: .public)")
         }
+
+        return data
     }
 
     func saveData(_ data: Data, for config: Configuration) throws {
         let file = fileUrl(for: config)
-        try data.write(to: file, options: .atomic)
+        var coordinatorError: NSError?
+
+        NSFileCoordinator().coordinate(writingItemAt: file, options: .forReplacing, error: &coordinatorError) { fileUrl in
+            do {
+                try data.write(to: fileUrl, options: .atomic)
+            } catch {
+                Logger.config.error("Unable to write \(config.rawValue, privacy: .public): \(error.localizedDescription, privacy: .public)")
+            }
+        }
+
+        if let coordinatorError {
+            Logger.config.error("Unable to write \(config.rawValue, privacy: .public): \(coordinatorError.localizedDescription, privacy: .public)")
```

Comment:
I notice that we have a pixel for read failures, but nothing for write failures. Though it looks like we never had one for write failures anyway, which seems strange - the previous code used a `try`, so maybe it gets caught elsewhere and a pixel is sent from there?

Reply:
I don't see a pixel being fired in these cases from anywhere else. But we'd probably have to triage it. Sounds like a follow up to me?